src/controllers/companyController.js

```diff
@@ -0,0 +1,87 @@
+import { companies } from "../models/index.js";
+import notFoundError from "../erros/notFoundError.js";
+
+class CompanyController {
+  static async registerCompany(req, res, next) {
+    try {
+      const newCompany = await companies.create(req.body);
+      res.status(201).json({ message: "Loja criada!", company: newCompany });
+    } catch (err) {
+      next(err);
+    }
+  }
+
+  static async listCompanyById(req, res, next) {
+    try {
+      const id = req.params.id;
+      const foundCompany = await companies.findById(id);
+
+      if (foundCompany !== null) {
+        res.status(200).json(foundCompany);
+      } else {
+        next(new notFoundError("Id do livro não localizado"));
+      }
```

Comment:
Aqui é mais uma perfumaria, tá? Mas se quiser, podes simplificar esse código, tornando ele mais linear e menos aninhado com elses. Algo do tipo:

```suggestion
     if (!foundCompany) {
        // se a company não foi encontradao
        next(new notFoundError("Company not found")); // lança o erro
        return; // return vazio para 'quebrar' o fluxo
      }
      res.status(200).json(foundCompany); // caso contrário, retorna um status positivo
```
Outro ponto de atenção aqui é sempre prestar atenção na mensagem de erro retornada. Lembra que não estamos lidando com livros aqui nesse contexto, mas com companies. 😉

Reply:
Perfumaria que nada, ainda tô aprendendo aonde aplicar essas guard clauses, valeu!

Reply:
Já adicionei para outros métodos com verificações equivalentes.